File: callmgr/expiredCalls.php
<?php

require_once(dirname(__FILE__).'/libtechfunctions/misc.php');
$vid=10; //Vendor ID for tring is 3
$maxretry=10;

ltecho("Marking Expired Calls");
ltecho("-------------------");
$mydbcon = ltopendb();
$query="select tc.vid,tc.bid,tc.cid,tc.callparams,tc.phone,tc.tid,tc.retry from ToCall tc, Broadcasts b where tc.bid=b.bid and b.endDate < CURDATE()   AND tc.inprogress=0  ;";
ltecho($query);
$result = mysqli_query($mydbcon,$query);
while($row = mysqli_fetch_array($result)) {
  $cid=$row['cid'];
  $bid=$row['bid'];
  $vid=$row['vid'];
  $phone=$row['phone'];
  $retry=$row['retry'];
  $attempts=$maxretry-$retry;
  $insquery="insert into callStatus (bid,expired,phone,attempts) values (".$bid.",1,'".$phone."',".$attempts.");";
  ltecho($insquery);
  $insresult = mysqli_query($mydbcon,$insquery);
  $delquery="delete from ToCall where cid=".$cid;
  ltecho($delquery);
  $delquery = mysqli_query($mydbcon,$delquery);
}
?>

File: callmgr/expiredCalls.py
import MySQLdb
import datetime
import os
import time
from settings import dbhost,dbuser,dbpasswd,sid,token


def main():
  finalCallStatus='expired'
  duration=0
  db = MySQLdb.connect(host=dbhost, user=dbuser, passwd=dbpasswd, charset='utf8')
  cur=db.cursor()
  db.autocommit(True)
  query="SET NAMES utf8"
  cur.execute(query)
  query="use libtech"
  cur.execute(query)
  query="select c.id,c.vendor,c.bid,c.phone,c.retry from callQueue c,broadcasts b  where b.bid=c.bid and b.endDate < CURDATE() and c.inprogress=0 "
  cur.execute(query)
  results = cur.fetchall()
  for row in results:
    callid=row[0] 
    vendor=row[1]
    bid=str(row[2])
    phone=row[3]
    retry=row[4]
    query="update callStatus set status='"+finalCallStatus+"',attempts="+str(retry)+",vendor='"+vendor+"',duration="+str(duration)+" where  bid="+bid+" and phone='"+phone+"';"
    print query
    cur.execute(query)
    query="delete from callQueue where id="+str(callid)
    print query
    cur.execute(query)

if __name__ == '__main__':
  main()
